src/query/query_manager.h

```diff
@@ -114,7 +114,8 @@ typedef enum
 typedef enum
 {
   QUERY_IN_PROGRESS,
-  QUERY_COMPLETED
+  QUERY_COMPLETED,
+  QUERY_ENDED
```

Comment:
QUERY_COMPLETED is set at query execution, but QUERY_ENDED means that the query result can be removed from cache (in case that transaction is aborted or cursor is not holdable).
query execution complete = QUERY_COMPLETED (query is completed but the cursor is not closed yet)
query end = QUERY_ENDED (at cursor close or aborted)

Reply:
The name might be confused. how about QUERY_CLOSED?

Reply:
good idea.